src/main/java/uk/gov/pay/products/persistence/dao/ProductDao.java

```diff
@@ -116,4 +116,32 @@ private List<ProductUsageStat> findProductsAndUsageQuery(Integer gatewayAccountI
         }
         return queryBuilder.getResultList();
     }
+
+    public List<ProductUsageStat> findUnusedProducts(Integer gatewayAccountId) {
+        return findUnusedProductsQuery(gatewayAccountId);
+    }
+
+    public List<ProductUsageStat> findUnusedProductsQuery(Integer gatewayAccountId) {
+        boolean shouldFilterGatewayAccount = gatewayAccountId != null;
+
+        String conditionalFilter = shouldFilterGatewayAccount ?
+                "AND product.gateway_account_id = ?1 " :
+                "";
+
+        String query = "SELECT product.* FROM products product " +
+                "LEFT JOIN payments payment ON product.id = payment.product_id " +
+                "WHERE payment.product_id IS NULL " +
+                conditionalFilter;
+
+        var queryBuilder = entityManager.get()
+                .createNativeQuery(query, ProductEntity.class);
+
+        if (shouldFilterGatewayAccount) {
+            queryBuilder = queryBuilder.setParameter(1, gatewayAccountId);
```

Comment:
make this a named param for clarity
```suggestion
        queryBuilder = queryBuilder.setParameter("gatewayAccountId", gatewayAccountId);
```

Reply:
named parameters aren't supported for native queries